src/components/MultiSelect/MultiSelectListbox.tsx

```diff
@@ -0,0 +1,135 @@
+    stateReducer: (state, actionAndChanges) => {
+      const { changes, type } = actionAndChanges;
+      switch (type) {
+        case useSelect.stateChangeTypes.MenuKeyDownEnter:
+        case useSelect.stateChangeTypes.MenuKeyDownSpaceButton:
+        case useSelect.stateChangeTypes.ItemClick:
+          return {
+            ...changes,
+            isOpen: true, // Keep menu open after selection.
+            highlightedIndex: state.highlightedIndex,
+          };
+        default:
+          return changes;
+      }
+    },
+    // @ts-ignore
+    selectedItem: selectedItems,
+    onSelectedItemChange: ({ selectedItem }) => {
+      onChange(selectedItem, id);
+    },
+    initialIsOpen: defaultIsOpen,
+  });
+
+  return (
+    <Box id={id} __css={styles}>
+      <MultiSelectMenuButton
+        multiSelectId={id}
+        label={label}
+        isOpen={isOpen}
+        selectedItems={selectedItems}
+        onClear={onClear}
+        {...getToggleButtonProps({
+          // @TODO add short prefix to avoid id starting with number
+          id: `${id}-toggle-button`,
+          "aria-labelledby": `${id}-label ${id}-toggle-button`,
+        })}
+      />
+      <Box __css={styles.menuContainer} {...(!isOpen && { display: "none" })}>
+        <UnorderedList
+          styleType="none"
+          marginInlineStart="0"
+          {...getMenuProps({
+            // @TODO add short prefix to avoid id starting with number
+            id: `${id}-menu`,
+            "aria-labelledby": `${id}-label`,
+          })}
+          // @FIX This prevents the menu from closing when checkbox or label is clicked.
+          onClick={(e) => e.preventDefault()}
+          __css={styles.menu}
+        >
+          {isOpen &&
+            items.map((item: MultiSelectItem, index: number) => (
+              <ListItem
+                py={1}
+                // @TODO fix this, we want to pass the key prop as part of ...getItemProps but get
+                // error  Missing "key" prop for element in iterator  react/jsx-key
+                key={item.id}
+                {...getItemProps({
+                  // @TODO add short prefix to avoid id starting with number
+                  id: `${id}-item-${index}`,
+                  key: item.id,
+                  item,
+                  index,
+                })}
+                style={
+                  highlightedIndex === index
+                    ? { backgroundColor: "#bde4ff" }
+                    : {}
+                }
+              >
+                <Checkbox
+                  id={item.id}
+                  labelText={item.name}
+                  showLabel={true}
+                  name={item.name}
+                  isChecked={
+                    selectedItems[id]?.items.find(
+                      (selectedItemId: string) => selectedItemId === item.id
+                    )
+                      ? true
+                      : false
```

Comment:
This is not necessary but you can shorten this by doing 
```
isChecked={
  !!selectedItems[id]?.items.find(
    (selectedItemId: string) => selectedItemId === item.id
  )
}
```
This will force the returned value from `.find` to be a boolean value.

You can also use `.includes` instead of `.find` but it's not necessary:
```
selectedItems[id]?.items.includes(item.id)
```

Reply:
fixed